### ISSUE TITLE
Merge base and variant context variables

### ISSUE THREAD

**Issue by fabianmarz** [netzstrategen/drupal-twig_fractal#3]
It's necessary to duplicate variables in the base and the variant context. See the `class` key of the following example:

```yml
context:
  text: Default
  tag: button
  class: button
variants:
  - name: Primary
    context:
      text: Primary
      tag: button
      class: button button--primary
```
As the variant is a modifier of the base component it would be more convenient to declare the modifier class only. Also the tag should not be duplicated but declared once. 

```yml
context:
  text: Default
  tag: button
  class: button
variants:
  - name: Primary
    context:
      text: Primary
      class: button--primary
```

**Proposed solution:**
1. Assign missing context keys from the base component to the variant
2. Merge context keys of the base component and the variant (variant should come last), except the text key
3. Strip duplicate values from the array

**Comment by sun** [comment]
> declare the modifier class only

Does Fractal's JS library understand this concept as well?  Or can we adjust it to do so?

If not, our components would only work within Drupal but not Fractal, no?

---
Initial idea for class-level docs:
```
In addition, it understands BEM-style modifiers in requested template
names (such as `button--primary`) and automatically merges the
variant-specific variables (`--primary`) into the component's default
variables (`button`).
```

**Comment by sun** [comment]
With regard to merging of values, I'd expect this change to not modify the currently existing basic array_replace behavior for all keys, but instead, introduce special behaviors for certain key names only.

At this point, I only see a special behavior for the `class` variables, which should be split by spaces and be merged instead of getting replaced.

I'd expect that same additive behavior to be also applied for a custom `class` variable specified in the calling template/code.

In other words, I should never be able to replace/remove any of the default class names defined in the component.

**Comment by fabianmarz** [comment]
As far as I can see Fractal is not aware of this, and therefore needs also some action. I'll come back to this later.

**Comment by sun** [comment]
Related background info:

frctl/fractal#135 removed merging of properties that are arrays in the context variables, because some users did not expect that behavior, but the entire arrays to get replaced instead.

Before that PR, array values were intentionally merged as lists; i.e., our `context.attributes.class` of a variant would be correctly merged into the component's default context variables.

However, that behavior would have only worked, if `context.attributes.class` was actually an array.  We most commonly use strings in our components:

```yml
context:
  attributes:
    class: button

variants:
  - modifier: primary
    name: Primary
    context:
      attributes:
        class: button--primary
```

This is why both our Fractal Drupal/Twig adapter (plus netzstrategen/twig-drupal#1) as well as our Fractal integration module for Drupal here (plus #5) implement special merge operations for context variables and attributes now.
